src/NHibernate/Cfg/Configuration.cs

```diff
@@ -171,7 +171,9 @@ protected void Reset()
 			propertyReferences = new List<Mappings.PropertyReference>();
 			FilterDefinitions = new Dictionary<string, FilterDefinition>();
 			interceptor = EmptyInterceptor.Instance;
+#pragma warning disable 618
 			properties = Environment.Properties;
```

Comment:
For fully preparing its removal, the code here should read the configuration file. (Or eventually cache statically in `Configuration` that file read result as a dictionary.)

Reply:
I decided to split this into several PRs. This deserves full re-implementation.